pkg/util/archive/archive.go

```diff
@@ -58,27 +58,50 @@ func Untar(tarball, targetDir string) error {
 
 	for {
 		header, err := tarReader.Next()
-		if err == io.EOF {
-			break
-		} else if err != nil {
+		switch {
```

Comment:
@hferentschik yes the way we implemented it first time only thinking about our tarball structure which had files in first level but when I tried same implementation it break when files are underneath of a directory structure.